api/v1alpha1/project_config_types.go

```diff
@@ -342,6 +345,147 @@ type AzureWebhookReceiverConfig struct {
 	SecretRef corev1.LocalObjectReference `json:"secretRef" protobuf:"bytes,1,opt,name=secretRef"`
 }
 
+// GenericWebhookReceiverConfig describes a generic webhook receiver
+// that can be configured to handle arbitrary webhook payloads using
+// a set of user-defined actions, targets, and match conditions using expressions and/or labels.
+type GenericWebhookReceiverConfig struct {
+	// SecretRef contains a reference to a Secret. For Project-scoped webhook
+	// receivers, the referenced Secret must be in the same namespace as the
+	// ProjectConfig.
+	//
+	// For cluster-scoped webhook receivers, the referenced Secret must be in the
+	// designated "cluster Secrets" namespace.
+	//
+	// The Secret's data map is expected to contain a `secret` key whose value is
+	// the shared secret used to authenticate the webhook requests sent by the
+	// generic webhook sender.
```

Comment:
This would be a bit more consistent with how we've documented this in other spots where this is the only authentication strategy in play. (DockerHub for example.)

```suggestion
	// The Secret's data map is expected to contain a `secret` key whose value
	// does NOT need to be shared directly with the sender. It is used only by
	// Kargo to create a complex, hard-to-guess URL, which implicitly serves as a
	// shared secret.
```